projects/laji/test/+vihko/trip-form-template.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { TripFormPage } from './trip-form.po';
+import { TemplatesView } from './tools.po';
+import { expect, Page, test } from '@playwright/test';
+import { login } from '../+user/user.po';
+import { ERROR_DIALOG_SELECTOR } from '../+error/error.po';
+
+test.describe('Trip form template page', () => {
+  test.describe.configure({ mode: 'serial' });
+
+  let page: Page;
+  let tripFormPage: TripFormPage;
+  let templatesView: TemplatesView;
+
+  test.beforeAll(async ({ browser }) => {
+    page = await browser.newPage();
+    await login(page);
+
+    tripFormPage = new TripFormPage(page);
+    templatesView = new TemplatesView(page);
+
+    await tripFormPage.navigateToTemplate();
+  });
+
+  test.afterEach(async () => {
+    await expect(page.locator(ERROR_DIALOG_SELECTOR)).not.toBeVisible();
+  });
+
+  test('template form is shown', async () => {
+    await tripFormPage.documentFormView.isTemplate();
+  });
+
+  test('should be able to fill in form with simple data', async () => {
+    await tripFormPage.fillInSimpleForm();
+    await expect(tripFormPage.$countryElem).toHaveValue('Suomi');
+  });
+
+
+  test('saving displays template naming form', async () => {
+    await tripFormPage.documentFormView.save();
+    await expect(tripFormPage.templateForm.$container).toBeVisible();
+  });
+
+  const name = Math.random().toString().substr(2, 8);
+
+  test('saving template directs to Vihko template page', async () => {
+    await tripFormPage.templateForm.$nameInput.fill(name);
+    await tripFormPage.templateForm.$descriptionInput.fill('test desc');
+    await tripFormPage.templateForm.$saveButton.click();
+    await expect(templatesView.$container).toBeVisible();
+  });
+
+  let row;
+
+  test('template is shown in table', async () => {
+    for (let idx = 0; idx <  await templatesView.datatable.$rows.count(); idx++) {
+      const iteratedRow = templatesView.datatable.getRow(idx);
+      const cell = iteratedRow.$container.locator(`span[title="${name}"]`);
+      if (await cell.isVisible()) {
+        row = iteratedRow;
+        break;
+      }
+    }
+    expect(!!row).toBe(true);
+  });
```

Comment:
This fails to me. I think one reason at least is that when the table is shown, it first shows a spinner before showing the rows. This doesn't use auto-retrying so it's flaky.

Not sure what would be the best approach... the row is needed for the tests after this one. At least the cell containing the name can be digged without iterating - just use the `span[title="${name}"]` locator. Then the delete button can be found by searching for the del button via the parent locator of the name cell. https://playwright.dev/docs/other-locators#parent-element-locator